staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go

```
@@ -293,7 +332,52 @@ func (a customResourceStrategy) GetAttrs(obj runtime.Object) (labels.Set, fields
 	if err != nil {
 		return nil, nil, err
 	}
-	return labels.Set(accessor.GetLabels()), objectMetaFieldsSet(accessor, a.namespaceScoped), nil
+	sFields, err := a.selectableFields(obj, accessor)
+	if err != nil {
+		return nil, nil, err
+	}
+	return accessor.GetLabels(), sFields, nil
+}
+
+// selectableFields returns a field set that can be used for filter selection.
+// This includes metadata.name, metadata.namespace and all custom selectable fields.
+func (a customResourceStrategy) selectableFields(obj runtime.Object, objectMeta metav1.Object) (fields.Set, error) {
+	objectMetaFields := objectMetaFieldsSet(objectMeta, a.namespaceScoped)
+	var selectableFieldsSet fields.Set
+
+	if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceFieldSelectors) && len(a.selectableFieldSet) > 0 {
+		us, ok := obj.(runtime.Unstructured)
+		if !ok {
+			return nil, fmt.Errorf("unexpected error casting a custom resource to unstructured")
+		}
+		uc := us.UnstructuredContent()
+
+		selectableFieldsSet = fields.Set{}
+		for _, sf := range a.selectableFieldSet {
+			if sf.err != nil {
+				return nil, fmt.Errorf("unexpected error parsing jsonPath: %w", sf.err)
+			}
+			results, err := sf.fieldPath.FindResults(uc)
+			if err != nil {
+				return nil, fmt.Errorf("unexpected error finding value with jsonPath: %w", err)
+			}
+			var value any
+
+			if len(results) > 0 && len(results[0]) == 1 {
```

if len(result) > 1, should we fail? that's unexpected, right?